test/hotspot/jtreg/runtime/cds/appcds/sharedStrings/InvalidFileFormat.java

```diff
@@ -26,6 +26,7 @@
  * @test
  * @summary Check most common errors in file format
  * @requires vm.cds.write.archived.java.heap
+ * @requires vm.flagless
```

Comment:
I think the reason you needed to add `vm.flagless` is SharedStringsUtils explicitly adds `-XX:UseG1GC` (long time ago this was required to dump the Java heap objects). It's better to remove `-XX:UseG1GC`.

You might need to ensure that all tests that use SharedStringsUtils have `@requires vm.cds.write.archived.java.heap`.

You should also check all other test cases to see if we misuse  `-XX:UseG1GC` for a similar reason.

Reply:
Looks like all the tests that use `SharedStringsUtil` already have `@requires vm.cds.write.archived.java.heap` so they should be fine. `SharedStringsHumongous` requires G1 so I'll leave that as-is which is also consistent with the Leyden repo.